scripts/recall-last-item.lua

```diff
@@ -22,19 +38,42 @@ local function on_player_cursor_stack_changed(e)
     return
   end
 
-  local item = util.get_cursor_item(player)
-  if not item then
+  if (e.tick == storage.item_update_tick[e.player_index]) then
     return
   end
 
-  storage.last_item[e.player_index] = item
+  local history = storage.item_history[e.player_index] or {}
```

Comment:
I would rather have this be an `if not history then (create default table and add to storage)`.